Security/Web_Security.mediawiki

```diff
@@ -0,0 +1,614 @@
+
+<tt>preload</tt> allows the website to be included in the [https://hstspreload.appspot.com/ HSTS preload list], upon submission. As a result, web browsers will do HTTPS upgrades to the site without ever having to receive the initial HSTS header.  This prevents downgrade attacks upon first use and is recommended for all high risk websites.  Note that being included in the HSTS preload list requires that <tt>includeSubDomains</tt> also be set.
+
+=== Examples ===
+
+<pre># Only connect to this site via HTTPS for the next six months
+Strict-Transport-Security: max-age=15768000</pre>
+
+<pre># Only connect to this site and subdomains via HTTPS for the next year and also include in the preload list
+Strict-Transport-Security: max-age=31536000; includeSubDomains; preload</pre>
+
+=== See Also ===
+
+* [https://developer.mozilla.org/docs/Web/Security/HTTP_strict_transport_security MDN on HTTP Strict Transport Security]
+* [https://tools.ietf.org/html/rfc6797 RFC6797: HTTP Strict Transport Security (HSTS)]
+
+
+== HTTP Redirections ==
+
+Websites may continue to listen on port 80 (HTTP) so that users do not get connection errors when typing a URL into their address bar, as browsers currently connect via HTTP for their initial request.  Sites that listen on port 80 should only redirect to the same resource on HTTPS. Once the redirection has occured, [[#HTTP Strict Transport Security|HSTS]] should ensure that all future attempts go to the site via HTTP are instead sent directly to the secure site. APIs or websites not intended for public consumption should disable the use of HTTP entirely.
+
+Redirections should be done with the 301 redirects, unless they redirect to a different path, in which case they may be done with 302 redirections. Sites should avoid redirections from HTTP to HTTPS on a different host, as this prevents HSTS from being set.
+
+=== Examples ===
+
+<pre># Redirect from http to https with nginx
+server {
+  listen 80;
+  server_name site.mozilla.org;
+
+  return 301 https://$server_name$request_uri;
+}</pre>
+
+<pre># Redirect from http to https with Apache
+<VirtualHost *:80>
+  ServerName site.mozilla.org
+  Redirect permanent / https://site.mozilla.org/
+</VirtualHost></pre>
+
+
+== HTTP Public Key Pinning ==
+
+Maximum risk sites must enable the use of HTTP Public Key Pinning (HPKP). HPKP instructs a user agent to bind a site to specific root certificate authority, intermediate certificate authority, or end-entity public key. This prevents  certificate authorities from issuing unauthorized certificates for a given domain that would nevertheless be trusted by the browsers. These fradulent certificates would allow an active attacker to MitM and impersonate a website, intercepting credentials and other sensitive data.
```

Comment:
l107 I would link to the RRA process for ex, so that they know how to determine what a maximum risk site is